### PR TITLE
refactor: more lang indirection and mutability removal

### PR DESCRIPTION
**Pull request by rockstar**
This is the final refactor patch that helped me understand the full interaction
here and what's going wrong with my current feature work. This patch removes
more indirection from functions that exist to only be called once, as well
`for` loops that required mutability and mutable parameters.

This patch seems scatter-brained, and it's true, to an extent. This area of the
code evolved in a kinda scatter-brained way, and my brain was also fried while
trying to figure this out. I'm confident, at this point, that I know the path
forward, and would like to take that path forward without littering it up with
the changes from this patch.

### CODE REVIEW

File: src/visitors/semantic/mod.rs [hunk 1]
```diff
@@ -195,20 +194,15 @@ pub struct ImportFinderVisitor {
...
                     .to_owned(),
             };
 
             self.imports.push(Import {
                 path: import.path.value.clone(),
-                alias,
-                initial_name: get_package_name(
-                    import.path.value.as_str(),
-                )
-                .map(|s| s.to_owned()),
```
Comment (appletreeisyellow) [inline src/visitors/semantic/mod.rs:211]: Are we replacing two entries `alias` and `initial_name` with one entry `name`? Just wondering why?

**Review by appletreeisyellow — APPROVED** [review]
I like the refactoring of indirect functions and `for` loops that makes flow clean

File: src/lang.rs [hunk 2]
```diff
@@ -73,83 +75,81 @@ fn record_fields(
...
+                    .map(|head| match &head.v {
+                        MonoType::Fun(f) => {
+                            Function::new(head.k.to_string(), f)
+                        }
+                        _ => unreachable!("Previous filter failed"),
+                    })
+                    .collect::<Vec<Function>>(),
+                _ => unreachable!("Previous filter failer"),
+            })
+            .collect()
```
[the remark this answers is not in the document]
Comment (rockstar) [reply src/lang.rs:99]: Rust (and, really, llvm) seems to be able to optimize iterators better than `for` loops. I'm not sure why that is, but it's just become the default action for me. I don't think I ever use `for` loops in rust anymore.

File: src/server/mod.rs [hunk 3]
```diff
@@ -982,14 +982,10 @@ impl LanguageServer for LspServer {
...
                         if let Some(env) = flux::imports() {
                             env.iter().filter(|(key, _val)| {
-                            if let Some(package_name) = lang::get_package_name(key) {
-                                fuzzy_match(package_name, &identifier.name)
-                            } else {
-                                false
-                            }
+                            fuzzy_match(lang::get_package_name(key), &identifier.name)
                         }).map(|(key, _val)| {
                             #[allow(clippy::unwrap_used)]
```
[the remark this answers is not in the document]
Comment (rockstar) [reply src/server/mod.rs:987]: Ah, good catch!

File: src/visitors/semantic/mod.rs (hunk 1 above)
Comment (rockstar) [reply src/visitors/semantic/mod.rs:211]: `alias` was never used, but something in the way the linter was looking at it, and I just changed `initial_name` to `name`

**Pull request merged**